framework/src/base/FEProblemBase.C

```diff
@@ -3132,9 +3130,17 @@ FEProblemBase::setCoupling(Moose::CouplingType type)
 void
 FEProblemBase::setCouplingMatrix(CouplingMatrix * cm)
 {
+  // TODO: Deprecate method
+
+  _coupling = Moose::COUPLING_CUSTOM;
+  _cm.reset(cm);
+}
+
+void
+FEProblemBase::setCouplingMatrix(std::unique_ptr<CouplingMatrix> cm)
+{
   _coupling = Moose::COUPLING_CUSTOM;
-  delete _cm;
-  _cm = cm;
+  _cm = std::move(cm);
```

Comment:
Never mind, after reading the article again it looks like by-value is the right way to go in this case, since the function *always* claims ownership in this case.